docs/architecture/lb/nginx.md

```diff
@@ -0,0 +1,270 @@
+    server {
+        listen        514 udp;
+        proxy_pass    stream_syslog_514;
+        
+        proxy_bind $remote_addr:$remote_port transparent;
+        proxy_responses 0;
+    }
+}
+```
+
+**For NGINX Plus:**
+
+1. Add the following configuration block to `/etc/nginx/nginx.conf`:
+```conf
+stream {
+    # Define upstream for each of SC4S hosts and ports
+    # Default SC4S UDP port is 514
+    # Include your custom ports if applicable
+    upstream stream_syslog_514 {
+        zone   stream_syslog_514 64k;
+        server <SC4S_IP_1>:514;
+        server <SC4S_IP_2>:514;
+    }
+
+    # Define connections to each of your upstreams.
+    # Ensure to include `proxy_bind` and `health_check`.
+    server {
+        listen        514 udp;
+        proxy_pass    stream_syslog_514;
+        
+        proxy_bind $remote_addr:$remote_port transparent;
+
+        health_check udp;
+    }
+}
+```
+
+NGINX will actively check the health of your upstream servers by sending UDP messages to port 514.
+
+2. (Optional) Add the following local post-filter to each of your SC4S instances to prevent SC4S from forwarding health check messages to Splunk and other destinations.
+`/opt/sc4s/local/config/app_parsers/nginx_healthcheck-postfiler.conf`
+```conf
+block parser nginx_healthcheck-postfiler() {
+    channel {
+        rewrite(r_set_dest_splunk_null_queue);
+    };
+};
+
+application nginx_healthcheck-postfiler[sc4s-postfilter] {
+    filter {
+        "${fields.sc4s_vendor}" eq "splunk" and
+        "${fields.sc4s_product}" eq "sc4s"
+        and message('nginx health check' type(string));
+    };
+    parser { nginx_healthcheck-postfiler(); };
+};
+```
+
+3. Refer to the NGINX documentation to find the command to reload the service, for example:
+```bash
+sudo nginx -s reload
+```
+
+4. Ensure that you disable `Source/Destination Checking` on your load balancer's host if you are working on AWS.
+
+### Test Your Setup
+1. Send UDP messages to the load balancer and verify that they are correctly received in Splunk with the correct host IP:
+```bash
+echo "hello world" > /dev/udp/<LB_IP>/514
+```
+
+2. Run performance tests
+
+| Receiver / Drops Rate for EPS (msgs/sec) | 4,500  | 9,000  | 27,000 | 50,000 | 150,000 | 300,000 |
+|------------------------------------------|--------|--------|--------|--------|---------|---------|
+| Single SC4S Server                       | 0.33%  | 1.24%  | 52.31% | 74.71% |    --   |    --   |
+| Load Balancer + 2 Servers                | 1%     | 1.19%  | 6.11%  | 47.64% |    --   |    --   |
+| Single Finetuned SC4S Server             | 0%     | 0%     | 0%     | 0%     |  47.37% |    --   |
+| Load Balancer + 2 Finetuned Servers      | 0.98%  | 1.14%  | 1.05%  | 1.16%  |  3.56%  |  55.54% |
+
+Please note that load balancer fine-tuning is beyond the scope of the SC4S team's responsibility. For assistance in minimizing UDP drops on the load balancer side, contact the NGINX support team.
```

Comment:
Note that load balancer support and fine-tuning is beyond the scope of the SC4S team's responsibility. For assistance in minimizing UDP drops on the load balancer side, contact your NGINX support team.